internal/it/test_skip.go

```diff
@@ -0,0 +1,221 @@
+			if err := validateLength(parts, 3, condition, "hz = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], HzVersion())
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipClientVersion:
+			if err := validateLength(parts, 3, condition, "ver = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], internal.ClientVersion)
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipOS:
+			if err := validateLength(parts, 3, condition, "os = windows"); err != nil {
+				return true, err
+			}
+			skip, err := checkOS(parts[1], parts[2])
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipEnterprise:
+			if err := validateLength(parts, 1, condition, "enterprise"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipEnterprise:
+			if err := validateLength(parts, 1, condition, "!enterprise"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipOSS:
+			if err := validateLength(parts, 1, condition, "oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipOSS:
+			if err := validateLength(parts, 1, condition, "!oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		default:
+			return true, fmt.Errorf("Unexpected test skip constant \"%s\" in %s", parts[0], condition)
+		}
+	}
+	return false, nil
+}
+
+func validateLength(parts []string, expected int, condition, example string) error {
+	if len(parts) != expected {
+		return fmt.Errorf("Unexpected format for %s, example of expected condition: \"%s\" ", condition, example)
+	}
+	return nil
+}
+
+func checkVersion(given, operator, actual string) (bool, error) {
+	isGivenGreater, err := compareVersions(given, actual)
+	if err != nil {
+		return true, err
+	}
+
+	switch operator {
+	case "=":
+		if isGivenGreater == nil {
+			return true, nil
+		}
+		return false, nil
+	case "!=":
+		if isGivenGreater != nil {
+			return true, nil
+		}
+		return false, nil
+	case ">":
+		if isGivenGreater != nil && !*isGivenGreater {
+			return true, nil
+		}
+		return false, nil
+	case ">=":
+		if isGivenGreater != nil && *isGivenGreater {
+			return false, nil
+		}
+		return true, nil
+	case "<":
+		if isGivenGreater != nil && *isGivenGreater {
+			return true, nil
+		}
+		return false, nil
+	case "<=":
+		if isGivenGreater != nil && !*isGivenGreater {
+			return false, nil
+		}
+		return true, nil
+	default:
+		return true, fmt.Errorf("Unexpected test skip operator \"%s\" to compare versions", operator)
```

Comment:
Instead of escaping the quote, you can use backticks: `Unexpected test skip operator %s`